openml/_api_calls.py

```diff
@@ -80,7 +81,7 @@ def _read_url_files(url, data=None, file_dictionary=None, file_elements=None):
     # 'gzip,deflate'
     response = requests.post(url, data=data, files=file_elements)
     if response.status_code != 200:
-        raise OpenMLServerError(('Status code: %d\n' % response.status_code) + response.text)
+        raise _parse_server_exception(response)
```

Comment:
You are correct. However, from a programming perspective, I would keep it the way it is. "_parse_server_exception" is an understandable function with expected inputs and outputs. The fact that the output is a raisable object, does not mean that the intended behaviour of this function should be raising an error.